includes/class-newspack-popups-model.php

```diff
@@ -344,10 +346,11 @@ public static function create_popup_object( $campaign_post, $include_taxonomies
 			'title'   => $campaign_post->post_title,
 			'content' => $campaign_post->post_content,
 			'options' => wp_parse_args(
-				array_filter( $post_options ),
```

Comment:
Why is the filtering removed? I think this served the purpose of allowing `wp_parse_args` to overwrite empty options.

Reply:
Because the `display_border` option is `true` by default, if you then toggle off the option the filtering overrides it, making it impossible to actually turn it off.

Reply:
At first I created the option as "Hide prompt border" with a default value of `false`, but it seemed weird to have all the other options phrased in the positive ("Display prompt title") with that option in the negative.

Reply:
The only change I've spotted is that without filtering `$post_options`, `dismiss_text_alignment`'s default value is not applied by `wp_parse_args`. This is not a problem though, because it defaults to 'center' anyway here:

https://github.com/Automattic/newspack-popups/blob/c8189dc9c6376fa89509bdf98b75e12f60c6f27f/includes/class-newspack-popups-model.php#L417-L419

However, this makes the `wp_parse_args` call misleading (not all defaults will be applied). This is not a big issue, but it'd be worth it to centralise the default values setting to `create_popup_object` method (without the need for stuff like `get_dismiss_text_alignment` or `get_dismiss_text`).

Reply:
I see what you mean how it's confusing that default values are set in multiple places. What about standardizing defining the defaults when the meta fields are registered? Tried it in 5f17039 and I think this lets us remove the `wp_parse_args` entirely as well as the getter methods such as `get_dismiss_text` and `get_dismiss_text_alignment` which previously defined defaults.

Reply:
Oh, I see what you mean about the defaults not being set, because the `retrieve_preview_popup` sets all of the options even if corresponding params aren't set on the URL. Let me revisit this.

Reply:
After struggling with this for a bit, I think it's not worth the extensive refactoring we'd need to ensure that a non-default `false` value is preserved over a default `true`. In ce04bce I restored the `array_filter` and flipped the `display_border` option back to `hide_border` with a default of `false`. But I think we can still remove the `get_dismiss_text` and `get_dismiss_text_alignment` methods since the defaults are already set with `wp_parse_args`.

Reply:
Yeah, that makes sense 👌